test/battle/hold_effect/metronome.c

```diff
@@ -33,8 +37,117 @@ SINGLE_BATTLE_TEST("Metronome gradually boosts power by 20%")
             HP_BAR(opponent, captureDamage: &damage[j]);
         }
     } THEN {
-        for (j = 1; j < METRONOME_TURNS; ++j) {
-            EXPECT_MUL_EQ(damage[0], MetronomeMultipliers[j-1], damage[j]);
+        for (j = 0; j < METRONOME_TURNS; ++j) {
+            EXPECT_MUL_EQ(damage[0], MetronomeMultipliers[j], damage[j]);
         }
     }
 }
+
+SINGLE_BATTLE_TEST("Metronome Item's boost is reset if the attacker uses a different move")
+{
+    s16 damage[2];
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET) { Item(ITEM_METRONOME); }
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_TACKLE); }
+        TURN { MOVE(player, MOVE_QUICK_ATTACK); }
+        TURN { MOVE(player, MOVE_TACKLE); }
```

Comment:
I think we need to add another Tackle before Quick Attack to make sure that a boost would've theorically happened.

Reply:
We already test for the boost happening in another test. The counter is 1 after the first Tackle, and gets reset only by using quick attack. (I.e. if quick attack was tackle, we would already see the boost applying)

Reply:
Fair, carry on.